src/parser/grammar.pegjs

```diff
@@ -141,63 +126,49 @@ non_equality_symbol
 //  Language parsing
 //
 
-
-language_predicate_arity
+//  START
+constants
+    = spaces c1:Identifier l:(spaces "," spaces c2:Identifier {return c2})* spaces {return [c1].concat(l)}
+predicates
+    = spaces p1:language_predicate l:(spaces "," spaces p2:language_predicate {return p2})* spaces {return [p1].concat(l)}
+functions
+    = spaces f1:language_function l:(spaces "," spaces f2:language_function {return f2})* spaces {return [f1].concat(l)}
+
+predicate_arity
     = $ ([0-9]+)
-
-language_function_arity
+function_arity
     = $ ([1-9]+)
 
-//  START
-// KONSTANTY
-// C
-
-language_constants_list
-    = spaces c1:Identifier c_list:(spaces "," spaces c2:Identifier {return c2})* spaces {return [c1].concat(c_list)}
-
-// START
-// PREDIKATY
-// P/2
-
-language_predicates_list
-    = spaces p1:language_predicate p_list:(spaces "," spaces p2:language_predicate {return p2})* spaces {return [p1].concat(p_list)}
-
 language_predicate
-    = spaces i:Identifier "/" arity:language_predicate_arity spaces {return {name: i, arity: arity}}
-
-// START
-// FUNKCIE
-// F/2
-
-language_functions_list
-    = spaces f1:language_function f_list:(spaces "," spaces f2:language_function {return f2})* spaces {return [f1].concat(f_list)}
-
+    = spaces predicate:Identifier "/" arity:predicate_arity spaces {return {name: predicate, arity: arity}}
 language_function
-    = spaces i:Identifier "/" arity:language_function_arity spaces {return {name: i, arity: arity}}
+    = spaces f:Identifier "/" arity:function_arity spaces {return {name: f, arity: arity}}
 
 
 //
 //  Structure parsing
 //
 
 // START
-// DOMENA
-
-structure_domain_items_list
-    = spaces i1:Identifier ids:(spaces "," spaces i2:Identifier {return i2})* spaces {return [i1].concat(ids)}
-
-// START
-// PREDIKAT + FUNKCIA
-
-structure_tuples_list
-    = spaces t1:structure_tuple tl:(spaces "," spaces t2:structure_tuple {return t2})* spaces {return [t1].concat(tl)}
+domain
+    = spaces i1:DomainIdentifier l:(spaces "," spaces i2:DomainIdentifier {return i2})* spaces {return [i1].concat(l)}
+tuples
+    = spaces t1:tuple l:(spaces "," spaces t2:tuple {return t2})* spaces {return [t1].concat(l)}
+e_tuples
+  = spaces t1:e_tuple l:(spaces "," spaces t2:e_tuple {return t2})* spaces {return [t1].concat(l)}
 
-structure_tuple
-    = "(" spaces item1:Identifier items:(spaces "," spaces item2:Identifier {return item2})+ spaces ")" {return checkTupleArity([item1].concat(items))}
-    / spaces item:Identifier spaces {return checkTupleArity(item)}
+tuple
+    = "(" spaces i1:DomainIdentifier l:(spaces "," spaces i2:DomainIdentifier {return i2})+ spaces ")" {return [i1].concat(l)}
+    / spaces i:DomainIdentifier spaces {return [i]}
+e_tuple
+    = "(" spaces i1:Identifier spaces "," spaces i2:DomainIdentifier spaces ")" {return [i1, i2]}
 
 
+DomainIdentifier
+    = name:IdentifierPart+ {return name.join("")}
```

Comment:
Zdá sa mi, že úplne najlepšie bude za prvok domény prehlásiť reťazec hocijakých symbolov, ktoré nepokazia parsovanie interpretácií funkcií a predikátov, teda ľubovoľný znak okrem: medzier (tých zo `spaces`, ale aj zo `Zs`), čiarky `,` a zátvoriek `(` a `)`.

To pokryje nie len emojis, ale aj všelijaké zaujímavé symboly (šachové ♚, ♛, astronomické ☉,☿, ☄︎, kartové ♠︎♣︎♥︎♦︎, matematické, …).